src/reporters/LambdaReporter.ts

```diff
@@ -0,0 +1,104 @@
+import { HttpClient } from "@actions/http-client";
+import { throws } from "assert";
+import { CheckOptions } from "../check";
+import { Result } from "../ComplianceChecker";
+import { Product } from "../model/Product";
+import { Reporter } from "./Reporter";
+
+interface ResultRecord {
+  level: string,
+  rule?: string,
+  check?: string,
+  mandatory?: string,
+  result: string,
+  reason?: string
+}
+
+interface RequestBody {
+  repo_owner?: string,
+  repo_name?: string,
+  start_time?: string,
+  end_time?: string,
+  result?: ResultRecord[]
+}
+
+export class LambdaReporter extends Reporter {
+
+  private readonly uri: string;
+  private readonly keyHeader: any;
+  private readonly http: HttpClient;
+  private body: RequestBody;
+
+  constructor() {
+    super();
+    if (process.env.INPUT_ENG_COMPLIANCE_LAMBDA_URI == null) throw new Error('INPUT_ENG_COMPLIANCE_LAMBDA_URI must be specified');
+    if (process.env.INPUT_ENG_COMPLIANCE_LAMBDA_KEY == null) throw new Error('INPUT_ENG_COMPLIANCE_LAMBDA_KEY must be specified');
+
+    this.uri = process.env.INPUT_ENG_COMPLIANCE_LAMBDA_URI as string;
+    this.body = {};
+    this.http = new HttpClient('eng-compliance-github');
+
+    let key = process.env.INPUT_ENG_COMPLIANCE_LAMBDA_KEY as string;
+    this.keyHeader = {
+      ['x-api-key']: key
+    }
+  }
+
+  static enabled(): boolean {
+    return process.env.INPUT_ENG_COMPLIANCE_LAMBDA_URI != null && process.env.INPUT_ENG_COMPLIANCE_LAMBDA_KEY != null;
+  }
+
+  startRun(product: Product) {
+    this.body = {
+      repo_owner: product.repo.owner,
+      repo_name: product.repo.name,
+      start_time: Date.now().toString(),
+      result: []
+    };
+  }
+
+  reportCheck(ruleName: string, checkName: string, checkOptions: CheckOptions, outcome: Result, message?: string) {
+    let check_record: ResultRecord = {
```

Comment:
<details>
<summary>:warning: <strong>This code contains an identifier named using non-standard conventions.</strong> (click for details)</summary>

A possible improvement would be to follow standard JavaScript naming conventions.  
  
This change would ensure the code follows well-known standards, making it easier for new developers to start without the need to learn a different convention and, thus, improving maintainability.

**[More details](https://confluence.devfactory.com/x/sHTlFQ)**

## The improvement suggestion in this case
```ts  
let checkRecord: ResultRecord = {
```  
> NOTE: Consider applying the structure of this solution in other parts of the code, if there is another occurrence(s).
</details>

[//]: # (<a id="a300e8df-2c77-4676-9e8f-b75a2c9dcc3f" data-finding-id="367359152" href="none" codereview-uuid-reserved-tag data-business-id="55ce652a-40cf-48e1-916a-f8d32a3b8762" data-analyzer="js" />)